lib/elixir/lib/string_io.ex

```diff
@@ -263,35 +255,24 @@ defmodule StringIO do
 
   ## get_line
 
-  defp get_line(encoding, prompt,
-                %{input: input, output: output, capture_prompt: capture_prompt} = s) do
-    case :unicode.characters_to_list(input, encoding) do
-      {:error, _, _} ->
-        {{:error, :collect_line}, s}
-      {:incomplete, _, _} ->
-        {{:error, :collect_line}, s}
-      chars ->
-        {result, input} = do_get_line(chars, encoding)
-
-        s =
-          if capture_prompt do
-            %{s | output: <<output::binary, IO.chardata_to_string(prompt)::binary>>}
-          else
-            s
-          end
-
-        {result, %{s | input: input}}
+  defp get_line(encoding, prompt, %{input: input} = s) do
+    case bytes_until_eol(input, encoding, 0) do
+      {:ok, count, replace_cr} ->
+        {result, remainder} = case count do
```

Comment:
We should try to avoid a nested case, especially when it is not required because it makes the code less clear. I think we could add `{:ok, 0, false}` pattern to top level case to handle this one. Note that we don't need to have a 3-tuple we could have `{:split, count} | {:replace_split, count} | :error`.